python/lsst/ctrl/bps/wms/htcondor/lssthtc.py

```diff
@@ -255,6 +259,64 @@ def __str__(self):
         return str(self.data)
 
 
+def htc_backup_files(wms_path, root=None, limit=100):
+    """Make backup copies of select HTCondor files in a given submit directory.
+
+    Files will be saved in separate subdirectories which will be created in
+    the submit directory where the files are located. These subdirectories
+    will be consecutive, zero-padded integers.
+
+    Parameters
+    ----------
+    wms_path : `str`
+        Path to the submission directory either absolute or relative.
+    root : `str`, optional
+        A top-level directory where all subdirectories with backup files will
+        be kept. Defaults to None which means that the backup subdirectories
+        will be placed directly in the submit directory.
+    limit : `int`, optional
+        Maximal number of backups. If the number of backups reaches the limit,
+        the last backup files will be overwritten. The default value is 100.
+
+    Raises
+    -------
+    FileNotFoundError
+        If the submit directory of the file that needs to be backed up does not
+        exist.
+    RuntimeError
+        If backing up a file failed due to filesystem-related issues.
+    """
+    width = len(str(limit))
+
+    path = Path(wms_path).resolve()
+    if not path.is_dir():
+        FileNotFoundError(f"Directory {path} not found")
+
+    # Initialize the backup counter.
+    runs = list(Path(wms_path).glob("*.rescue*"))
+    counter = max(0, min(len(runs) - 1, limit))
```

Comment:
This counter was supposed to work in terms of workflow restarts. So for the first run attempt (no rescue DAG if still ongoing or one rescue DAG if it already failed) it should give you 0. However, your comment made me realize that it is flawed anyway (it will be 0 even after the first restart if the new run hasn't finished yet). Will fix.